enterprise/models.py

```diff
@@ -4692,13 +4692,17 @@ def _get_filtered_ecu_ids(self, user_query):
         # https://docs.djangoproject.com/en/5.0/topics/security/
         var_q = f"%{user_query}%"
         sql_string = """
-            select entcu.id from enterprise_enterprisecustomeruser entcu
-            join auth_user au on entcu.user_id = au.id
-            where entcu.enterprise_customer_id = %s and au.email like %s;
+            with users as (
+                select ecu.id, au.email, coalesce(aup.name, concat(au.first_name, ' ', au.last_name)) as full_name
```

Comment:
Also I was testing on stage with a combination of set profile.name and user.first_name and last_name and I noticed a problem with how this query is working. It fails the test case of having to fall back to first/last from auth_user and never does, but I fixed it with this NULLIF function.                     
```coalesce(NULLIF(aup.name, ''), concat(au.first_name, ' ', au.last_name)) as full_name```